server/api/Products.js

```diff
@@ -1,46 +1,89 @@
 const {
-  models: { Product },
+  models: { Product, Room },
 } = require('../db/index');
 const router = require('express').Router();
 
 // Select all products
 
-router.get('/', async (req, res, next) => {
+router.get('/all', async (req, res, next) => {
   try {
     allProducts = await Product.findAll();
-
     res.status(200).send(allProducts);
+  } catch (err) {
+    next(err);
+  }
+});
+
+// get with queries to paginate and filter
+
+router.get('/', async (req, res, next) => {
+  const { page, size, type, style, room, sort } = req.query;
+  const options = {
+    attributes: {
+      include: [
+        'id',
+        'name',
+        'material',
+        'color',
+        'imageUrl',
+        'price',
+        'type',
+        'style',
+        'availability',
+      ],
+    },
+    order: ['name'],
+    where: {},
+  };
+  if (page && size) {
+    options.offset = (page - 1) * size;
+    options.limit = size;
+  }
+  if (type) {
+    options.where.type = type;
+  }
+  if (style) {
+    options.where.style = style;
+  }
+  if (room) {
+    console.log(room);
+    const { id } = await Room.findOne({ where: { name: room } });
+    console.log(id);
+    options.where.roomId = id;
+  }
```

Comment:
For `room` filtering, we need to find the appropriate room's id and set it as `options.where.roomId`